presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveMixedCaseSupport.java

```diff
@@ -0,0 +1,218 @@
+     *
+     * Each test ensures the schema is created successfully.
+     */
+    @Test(groups = {MIXED_CASE})
+    public void testCreateSchemasWithMixedCaseNames()
+    {
+        // Define schema names and their expected stored values in Hive
+        String schemaNameMixedSyllables = "HiveMixedCaseOn";
+        String schemaNameMixed = "HiveMixedCase";
+
+        String schemaNameLowerStored = "hivemixedcaseon";
+        String schemaNameMixedSyllablesStored = "hivemixedcaseon";
+        String schemaNameMixedStored = "hivemixedcase";
+        String schemaNameUpperStored = "hivemixedcaseon1";
+
+        // Create schemas
+        query("CREATE SCHEMA IF NOT EXISTS " + SCHEMA_NAME);
+        query("CREATE SCHEMA IF NOT EXISTS " + schemaNameMixedSyllables);
+        query("CREATE SCHEMA IF NOT EXISTS " + schemaNameMixed);
+        query("CREATE SCHEMA " + SCHEMA_NAME_UPPER);
+
+        // Verify schema creation
+        assertThat(query("SHOW SCHEMAS"))
+                .contains(row(schemaNameLowerStored))
+                .contains(row(schemaNameMixedSyllablesStored))
+                .contains(row(schemaNameMixedStored))
+                .contains(row(schemaNameUpperStored));
+    }
+
+    /*
+     * Test cases for creating tables with different naming conventions in Hive.
+     *
+     * This test verifies table creation, schema behavior, and column definitions
+     * when using different case variations for table and column names.
+     *
+     * Scenarios covered:
+     * 1. Creating tables with lowercase, mixed-case, and uppercase names.
+     * 2. Creating tables with uppercase column names.
+     * 3. Creating tables in schemas with lowercase and uppercase names.
+     *
+     * The test ensures:
+     * - Tables are created successfully.
+     * - Table names are stored and retrieved correctly.
+     * - Column definitions match expected types.
+     */
+    @Test(groups = {MIXED_CASE})
+    public void testCreateTablesWithMixedCaseNames()
+    {
+        // Creating tables with different name conventions
+        query("CREATE TABLE " + SCHEMA_NAME + ".testtable0 (name VARCHAR(50), id INT)");
+        query("CREATE TABLE " + SCHEMA_NAME + ".testtable (name VARCHAR(50), ID INT)");
+        query("CREATE TABLE " + SCHEMA_NAME + ".TestTable1 (Name VARCHAR(50), id INT)");
+        query("CREATE TABLE " + SCHEMA_NAME + ".TESTTABLE2 (Name VARCHAR(50), ID INT)");
+        query("CREATE TABLE " + SCHEMA_NAME_UPPER + ".TESTTABLE4 (Name VARCHAR(50), ID INT)");
+        query("CREATE TABLE " + SCHEMA_NAME_UPPER + ".testtable02 (name VARCHAR(50), id INT, num DOUBLE)");
+        query("CREATE TABLE " + SCHEMA_NAME_UPPER + ".TESTTABLE3 (name VARCHAR(50), id INT, num DOUBLE)");
+
+        // Verify table existence and column properties
+        assertThat(query("SHOW TABLES FROM " + SCHEMA_NAME))
+                .containsOnly(row("testtable0"), row("testtable"), row("testtable1"), row("testtable2"));
```

Comment:
I thought we would expect table names to come back with the same casing used in creation?

Reply:
For hive, since there is no specific connector level normalization, so it would be default to lowecase.